internal/webhooks/clusterclass_test.go

```diff
@@ -1318,6 +1320,854 @@ func TestClusterClassValidationWithClusterAwareChecks(t *testing.T) {
+						Name: "cpu",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type: "integer",
+								// Maximum changed to a value that invalidates ClusterVariable.
+								Maximum: pointer.Int64(3),
+							},
+						},
+					}).
+				Build(),
+			expectErr: true,
+		},
+		{
+			name: "Error if Type ClusterClassVariable is changed (Integer -> String) invalidating an existing ClusterVariable",
+			clusters: []client.Object{
+				builder.Cluster(metav1.NamespaceDefault, "cluster1").
+					WithLabels(map[string]string{clusterv1.ClusterTopologyOwnedLabel: ""}).
+					WithTopology(
+						builder.ClusterTopology().
+							WithClass("class1").
+							WithVariables(
+								clusterv1.ClusterVariable{
+									Name: "cpu",
+									Value: apiextensionsv1.JSON{
+										Raw: []byte(`4`),
+									},
+								}).
+							Build()).
+					Build(),
+			},
+			oldClusterClass: clusterClassBuilder.
+				WithVariables(
+					clusterv1.ClusterClassVariable{
+						Name: "cpu",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type: "integer",
+							},
+						},
+					},
+				).
+				Build(),
+			newClusterClass: clusterClassBuilder.
+				WithVariables(
+					clusterv1.ClusterClassVariable{
+						Name: "cpu",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								// Type changed to a value that invalidates ClusterVariable.
+								Type: "string",
+							},
+						},
+					}).
+				Build(),
+			expectErr: true,
+		},
+		{
+			name: "Pass if Type ClusterClassVariable is changed (Integer -> Number) not invalidating an existing ClusterVariable",
+			clusters: []client.Object{
+				builder.Cluster(metav1.NamespaceDefault, "cluster1").
+					WithLabels(map[string]string{clusterv1.ClusterTopologyOwnedLabel: ""}).
+					WithTopology(
+						builder.ClusterTopology().
+							WithClass("class1").
+							WithVariables(
+								clusterv1.ClusterVariable{
+									Name: "cpu",
+									Value: apiextensionsv1.JSON{
+										Raw: []byte(`4`),
+									},
+								}).
+							Build()).
+					Build(),
+			},
+			oldClusterClass: clusterClassBuilder.
+				WithVariables(
+					clusterv1.ClusterClassVariable{
+						Name: "cpu",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type: "integer",
+							},
+						},
+					},
+				).
+				Build(),
+			newClusterClass: clusterClassBuilder.
+				WithVariables(
+					clusterv1.ClusterClassVariable{
+						Name: "cpu",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								// Type changed to a value that does not invalidate ClusterVariable.
+								Type: "number",
+							},
+						},
+					}).
+				Build(),
+			expectErr: false,
+		},
+		{
+			name: "Pass if Type ClusterClassVariable is changed ( Number-> Integer) not invalidating an existing ClusterVariable",
```

Comment:
Looks like this does not pass (as expected), so I think the name should be adjusted accordingly.